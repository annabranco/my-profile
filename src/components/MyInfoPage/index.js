import React from 'react';
import { useSelector } from 'react-redux';
import Social from '../Social';
import { finishedSelector, infoPageTextSelector } from '../../redux/selectors';
import { isFullWindowDesktop } from '../../utils/device';
import { Mugshot, Barquinho } from '../../images';
import {
  FirstArrowIcon,
  InfoArea,
  InfoMessage,
  JobTitle,
  LineOfArrows,
  MainAreaWrapper,
  MeuBarquinho,
  MyInfoInnerWrapper,
  MyInfoWrapper,
  MyNameOnDesktopDevices,
  MyNameOnMobileDevices,
  Photo,
  PhotoWrapper,
  ScrollDownDisplay,
  SecondArrowIcon,
  SectionMyInfo,
  SocialArea,
  ThirdArrowIcon
} from './styles';

const MyInfoPage = () => {
  const finishedScenario = useSelector(finishedSelector);
  const texts = useSelector(infoPageTextSelector);

  return (
    <SectionMyInfo id="My Info Section">
      <MyInfoWrapper>
        <MyNameOnMobileDevices>Anna Branco</MyNameOnMobileDevices>

        <MyInfoInnerWrapper>
          <MainAreaWrapper>
            <PhotoWrapper>
              <Photo src={Mugshot} alt="Anna Branco" />
            </PhotoWrapper>
            <InfoArea>
              <MyNameOnDesktopDevices>Anna Branco</MyNameOnDesktopDevices>
              <JobTitle>{texts.job}</JobTitle>
              <InfoMessage>
                {!finishedScenario ? texts.aditional : texts.thankyou}
              </InfoMessage>
            </InfoArea>
          </MainAreaWrapper>
          <SocialArea>
            <Social texts={texts} />
          </SocialArea>
        </MyInfoInnerWrapper>

        <MeuBarquinho src={Barquinho} alt="Navigating beautifully" />

        {!isFullWindowDesktop && (
          <MyInfoWrapper className="infoPage__advise">
            {texts.advise}
          </MyInfoWrapper>
        )}

        <ScrollDownDisplay>
          <LineOfArrows>
            {[0, 1, 2].map(index => (
              <FirstArrowIcon
                key={index}
                className="fas fa-angle-double-down"
              />
            ))}
          </LineOfArrows>
          <LineOfArrows>
            {[0, 1, 2].map(index => (
              <SecondArrowIcon
                key={index}
                className="fas fa-angle-double-down"
              />
            ))}
          </LineOfArrows>
          <LineOfArrows>
            {[0, 1, 2].map(index => (
              <ThirdArrowIcon
                key={index}
                className="fas fa-angle-double-down"
              />
            ))}
          </LineOfArrows>
        </ScrollDownDisplay>
      </MyInfoWrapper>
    </SectionMyInfo>
  );
};

export default MyInfoPage;
