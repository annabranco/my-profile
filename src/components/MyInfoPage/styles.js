import styled, { keyframes } from 'styled-components';
import { rgba } from 'polished';
import { Binary } from '../../assets/images';
import { colorBlueLight, fontSubtitle, fontTitle } from '../../styles/theme';
import { MainBackground, NotDisplayed } from '../../styles/global';

const Vanish = keyframes`
  0% {
    opacity : 0;
  }

  5% {
    opacity : 1;
  }
`;

const ArrowIcon = styled.i`
  opacity: 0;
  margin: 0 20px;
  font-size: 3rem;
  color: ${rgba(colorBlueLight, 0.55)};
  animation-name: ${Vanish};
  animation-duration: 3s;
  animation-iteration-count: infinite;
  animation-fill-mode: forwards;

  @media all and (min-width: 768px) {
    margin: 0 100px;
  }
`;

const MyName = styled.h1`
  background-image: ${`url(${Binary})`};
  background-position: center;
  background-size: contain;
  -webkit-background-clip: text;
  background-clip: text;
  font-family: ${fontTitle};
  font-size: 3rem;
  font-weight: 900;
  color: transparent;
  display: block;
  margin: 5px 0;
  text-align: center;

  @media all and (min-width: 768px) {
    font-size: 4rem;
  }

  @media all and (min-width: 1024px) {
    font-size: 5rem;
  }
`;

const Navigating = keyframes`
  0% {
    right   : 20vw;
    opacity : 0;
  }

  10% {
    opacity : 1;
  }

  50% {
    opacity : 1;
  }

  70% {
    right   : 60vw;
    opacity : 0;
  }

  100% {
    right   : 110vw;
    opacity : 0;
  }
`;

export const BirdsFlying = styled.img`
  position: absolute;
  top: 30%;
  transform: translate(-4%, 0);
  width: 200px;
  opacity: 0.3;
  animation-name: ${Navigating};
  animation-duration: 120s;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
  animation-fill-mode: forwards;
  filter: contrast(0.5) opacity(0.6);

  @media all and (min-width: 768px) {
    top: 20%;
  }
`;
BirdsFlying.displayName = 'BirdsFlying';

export const FirstArrowIcon = styled(ArrowIcon)`
  animation-delay: 2s;
`;
FirstArrowIcon.displayName = 'Animated Arrows';

export const InfoArea = styled.div`
  height: 230px;
  width: 100%;
  padding: 10px 20px 10px 40px;

  @media all and (min-width: 500px) {
    height: 400px;
    width: 70%;
  }
`;
InfoArea.displayName = 'Info Area';

export const InfoMessage = styled.div`
  margin: -20px 0 0;
  width: 90%;
  font-style: italic;
  font-family: ${fontSubtitle};
  font-size: 0.8rem;
  line-height: 1.5;
  color: ${colorBlueLight};
  text-align: center;

  @media all and (min-width: 500px) {
    margin: 20px 0 0 20px;
    height: 200px;
    width: 80%;
    font-size: 1.2rem;
    text-align: left;
  }
`;
InfoMessage.displayName = 'Info Message';

export const JobTitle = styled.h2`
  margin-top: -30px;
  margin-left: -25px;
  width: 42vw;
  font-family: ${fontSubtitle};
  font-size: 1.3rem;
  text-align: center;

  @media all and (min-width: 321px) {
    margin-top: 0;
    font-size: 1.5rem;
  }

  @media all and (min-width: 500px) {
    margin-left: 0;
    width: auto;
    font-size: 2rem;
    text-align: left;
  }
`;
JobTitle.displayName = 'Job Title';

export const LineOfArrows = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
LineOfArrows.displayName = 'Line of Arrows';

export const MainAreaWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-wrap: wrap;
  width: 45%;

  @media all and (min-width: 500px) {
    width: 75%;
  }
`;
MainAreaWrapper.displayName = 'Main Area';

export const MeuBarquinho = styled.img`
  position: absolute;
  top: 40%;
  filter: saturate(0.5) brightness(1.8) contrast(30%) opacity(0.2);
  transform: translate(-50%, 0);
  width: 100px;
  animation-name: ${Navigating};
  animation-duration: 120s;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
  animation-fill-mode: forwards;

  @media all and (min-width: 768px) {
    top: 30%;
    filter: saturate(0.5) brightness(1.8) contrast(30%) opacity(0.2);
  }
`;
MeuBarquinho.displayName = 'Ship Image';

export const MyInfoInnerWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-wrap: wrap;
  margin: 0 auto;
  width: 95%;

  @media all and (min-width: 500px) {
    width: 90%;
  }
`;
MyInfoInnerWrapper.displayName = 'Inner Wrapper';

export const MyInfoWrapper = styled.div`
  margin-top: 50px;
  width: 100%;
  @media all and (min-width: 500px) {
    margin-top: 100px;
  }
`;
MyInfoWrapper.displayName = 'My Info Area';

export const MyNameOnDesktopDevices = styled(MyName)`
  ${NotDisplayed};
  text-align: left;
  @media all and (min-width: 500px) {
    display: block;
  }
`;
MyNameOnDesktopDevices.displayName = 'My Name';

export const MyNameOnMobileDevices = styled(MyName)`
  display: block;
  margin: 5px 0;
  text-align: center;

  @media all and (min-width: 500px) {
    ${NotDisplayed};
  }
`;
MyNameOnMobileDevices.displayName = 'My Name';

export const Photo = styled.img`
  width: 60%;
  border-radius: 50%;
  border: 1px solid green;
  box-shadow: 0 0 25px 1px ${rgba('green', 0.2)};
  filter: drop-shadow(2px 4px 6px rgba(0, 0, 0, 0.6)) contrast(0.9) sepia(0)
    brightness(1.3); ;
`;
Photo.displayName = 'Photo';

export const PhotoWrapper = styled.div`
  height: 170px;
  width: 100%;
  padding-top: 25px;
  height: 500px;
  display: flex;
  align-items: flex-start;
  justify-content: center;

  @media all and (min-width: 500px) {
    height: 400px;
    width: 30%;
  }
`;
PhotoWrapper.displayName = 'Photo Wrapper';

export const ScrollDownDisplay = styled.div`
  z-index: 3;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 0;

  @media all and (min-width: 768px) {
    margin-top: 80px;
  }
`;
ScrollDownDisplay.displayName = 'ScrollDown Display';

export const SecondArrowIcon = styled(ArrowIcon)`
  animation-delay: 3s;
`;
SecondArrowIcon.displayName = 'Animated Arrows';

export const SectionMyInfo = styled.section`
  ${MainBackground}
`;
SectionMyInfo.displayName = 'MY INFO Section';

export const SocialArea = styled.div`
  height: 400px;
  width: 55%;

  @media all and (min-width: 321px) {
    width: 50%;
  }

  @media all and (min-width: 500px) {
    width: 25%;
  }
`;
SocialArea.displayName = 'Social Area';

export const ThirdArrowIcon = styled(ArrowIcon)`
  animation-delay: 4s;
`;
ThirdArrowIcon.displayName = 'Animated Arrows';
