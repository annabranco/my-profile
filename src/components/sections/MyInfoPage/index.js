import React from 'react';
import { useSelector } from 'react-redux';
import { func } from 'prop-types';
import Social from '../../elements/Social';
import {
  finishedSelector,
  infoPageTextSelector,
  secionsTextsSelector
} from '../../../redux/selectors';
import { isDesktop, isFullWindowDesktop } from '../../../utils/device';
import {
  Mugshot,
  Barquinho,
  Birds,
  BlinkingFish
} from '../../../assets/images';
import {
  Advise,
  BirdsFlying,
  InfoArea,
  InfoMessage,
  JobTitle,
  LineOfArrows,
  MainAreaWrapper,
  MeuBarquinho,
  MoreText,
  MyInfoInnerWrapper,
  MyInfoWrapper,
  MyNameOnDesktopDevices,
  MyNameOnMobileDevices,
  Photo,
  PhotoWrapper,
  ScrollDownDisplay,
  SectionMyInfo,
  SocialArea,
  KindFish,
  ArrowIcon
} from './styles';
import { SKILLS_SECTION } from '../../../constants';

const MyInfoPage = ({ changeActiveSection }) => {
  const finishedScenario = useSelector(finishedSelector);
  const texts = useSelector(infoPageTextSelector);
  const sections = useSelector(secionsTextsSelector);
  const arrowLines = isDesktop ? 3 : 1;

  return (
    <SectionMyInfo id="My Info Section">
      <MyInfoWrapper>
        <MyNameOnMobileDevices>Anna Branco</MyNameOnMobileDevices>

        <MyInfoInnerWrapper>
          <MainAreaWrapper>
            <PhotoWrapper>
              <Photo src={Mugshot} alt="Anna Branco" />
            </PhotoWrapper>
            <InfoArea>
              <MyNameOnDesktopDevices>Anna Branco</MyNameOnDesktopDevices>
              <JobTitle>{texts.job}</JobTitle>
              {finishedScenario && <KindFish src={BlinkingFish} />}
              <InfoMessage>
                {!finishedScenario ? texts.aditional : texts.thankyou}
              </InfoMessage>
            </InfoArea>
          </MainAreaWrapper>
          <SocialArea>
            <Social texts={texts} />
          </SocialArea>
        </MyInfoInnerWrapper>

        <BirdsFlying src={Birds} />
        <MeuBarquinho src={Barquinho} alt="Navigating beautifully" />

        {!isFullWindowDesktop && (
          <Advise className="infoPage__advise">{texts.advise}</Advise>
        )}

        <ScrollDownDisplay>
          {!isDesktop && (
            <MoreText
              onClick={() => {
                console.log('ckick');
                changeActiveSection(SKILLS_SECTION);
              }}
            >
              {sections.technical}
            </MoreText>
          )}
          {[...Array(arrowLines).keys()].map(line => (
            <LineOfArrows key={line}>
              {[0, 1, 2].map(index => (
                <ArrowIcon
                  key={`${line}-${index}`}
                  className="fas fa-angle-double-down"
                />
              ))}
            </LineOfArrows>
          ))}
        </ScrollDownDisplay>
      </MyInfoWrapper>
    </SectionMyInfo>
  );
};

MyInfoPage.propTypes = {
  changeActiveSection: func.isRequired
};

export default MyInfoPage;
