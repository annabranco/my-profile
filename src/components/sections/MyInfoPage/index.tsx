import React, { ReactElement } from 'react';
import { useSelector } from 'react-redux';
import { AppState, InfoPageTextType } from '../../../types/interfaces';
import Social from '../../elements/Social';
import {
  finishedSelector,
  infoPageTextSelector
} from '../../../redux/selectors';
import { isDesktop, isFullWindowDesktop } from '../../../utils/device';
import {
  Barquinho,
  Birds,
  BlinkingFish,
  Mugshot
} from '../../../assets/images';
import {
  Advise,
  ArrowIcon,
  BirdsFlying,
  InfoArea,
  InfoMessage,
  JobTitle,
  KindFish,
  LineOfArrows,
  MainAreaWrapper,
  MeuBarquinho,
  MyInfoInnerWrapper,
  MyInfoWrapper,
  MyNameOnDesktopDevices,
  MyNameOnMobileDevices,
  Photo,
  PhotoWrapper,
  SectionMyInfo,
  SocialArea
} from './styles';

const MyInfoPage = (): ReactElement => {
  const finishedScenario: AppState['finishedScenario'] = useSelector(
    finishedSelector
  );
  const texts: InfoPageTextType = useSelector(infoPageTextSelector);
  const arrowLines: number = isDesktop ? 3 : 1;

  return (
    <SectionMyInfo id="My Info Section">
      <MyInfoWrapper>
        <MyNameOnMobileDevices>Anna Branco</MyNameOnMobileDevices>

        <MyInfoInnerWrapper>
          <MainAreaWrapper>
            <PhotoWrapper>
              <Photo src={Mugshot} alt="Anna Branco" />
            </PhotoWrapper>
            <InfoArea>
              <MyNameOnDesktopDevices>Anna Branco</MyNameOnDesktopDevices>
              <JobTitle>{texts.job}</JobTitle>
              {finishedScenario && <KindFish src={BlinkingFish} />}
              <InfoMessage>
                {!finishedScenario ? texts.aditional : texts.thankyou}
              </InfoMessage>
            </InfoArea>
          </MainAreaWrapper>
          <SocialArea>
            <Social texts={texts} />
          </SocialArea>
        </MyInfoInnerWrapper>

        <BirdsFlying src={Birds} />
        <MeuBarquinho src={Barquinho} alt="Navigating beautifully" />

        {!isFullWindowDesktop && (
          <Advise className="infoPage__advise">{texts.advise}</Advise>
        )}

        {isDesktop && (
          <>
            {[...Array(arrowLines).keys()].map(line => (
              <LineOfArrows key={line}>
                {isDesktop ? (
                  <>
                    {[0, 1, 2].map(index => (
                      <ArrowIcon
                        key={`${line}-${index}`}
                        className="fas fa-angle-double-down"
                      />
                    ))}
                  </>
                ) : (
                  <ArrowIcon className="fas fa-angle-double-down" />
                )}
              </LineOfArrows>
            ))}
          </>
        )}
      </MyInfoWrapper>
    </SectionMyInfo>
  );
};

export default MyInfoPage;
