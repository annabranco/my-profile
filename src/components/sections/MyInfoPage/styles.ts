import * as React from 'react';
import styled, { css, keyframes } from 'styled-components';
import { rgba } from 'polished';
import { Binary } from '../../../assets/images';
import {
  colorBlack,
  colorBlueDark,
  colorBlueLight,
  colorWhite,
  fontSubtitle,
  fontTitle,
  fontTitleAlt
} from '../../../styles/theme';
import {
  Appear,
  Disappear,
  MainBackground,
  NotDisplayed
} from '../../../styles/global';

const DownWeGo = (displacement: number) => keyframes`
  0%  {
    opacity : 0;
    transform: none;
  }

  5% {
    opacity : 1;
  }
  15%, 35%, 55% {
    opacity : 0;
  }
  20% {
    opacity : 1;
    transform: translate(0,${displacement}px);
  }
  40% {
    opacity : 1;
    transform: translate(0,${displacement * 2}px);
  }
  60% {
    opacity : 1;
    transform: translate(0,${displacement * 3}px);
  }
  75% {
    opacity : 0;
    transform: translate(0,${displacement * 3}px);
  }
`;

const MyName = styled.h1`
  background-image: ${`url(${Binary})`};
  background-position: center;
  background-size: contain;
  -webkit-background-clip: text;
  background-clip: text;
  font-family: ${fontTitle};
  font-size: 3rem;
  font-weight: 700;
  color: transparent;
  display: block;
  margin: 5px 0;
  text-align: center;

  @media all and (min-width: 768px) {
    font-size: 4rem;
  }

  @media all and (min-width: 1024px) {
    font-size: 5rem;
  }

  @media all and (min-width: 2000px) {
    font-size: 6rem;
  }
`;

const Navigating = keyframes`
  0% {
    right   : 20vw;
    opacity : 0;
  }

  10% {
    opacity : 1;
  }

  50% {
    opacity : 1;
  }

  70% {
    right   : 60vw;
    opacity : 0;
  }

  100% {
    right   : 110vw;
    opacity : 0;
  }
`;

export const Advise = styled.div`
  position: absolute;
  margin-left: 3vw;
  padding: 5px 10px;
  bottom: 20vh;
  width: 45vw;
  background: rgba(254, 255, 255, 0.3);
  font-size: 0.8em;
  text-align: left;
  color: ${colorBlack};
  font-family: ${fontSubtitle};
  opacity: 0;
  animation: ${Appear} 10s;
  animation-fill-mode: forwards;
`;
Advise.displayName = 'Advise';

interface ArrowIconProps extends React.ComponentPropsWithoutRef<'i'> {
  sections?: boolean;
  goBack?: boolean;
}

export const ArrowIcon = styled.i<ArrowIconProps>`
  margin: 0 20px;
  font-size: ${({ sections }) => (sections ? '1.2rem' : '1.5rem')};
  color: ${rgba(colorWhite, 0.55)};
  transform: rotate(270deg);

  ${({ goBack }) =>
    goBack &&
    css`
      transform: rotate(180deg);
    `}

  @media all and (min-width: 768px) {
    margin: 0 100px;
    font-size: 3rem;
    color: ${rgba(colorBlueLight, 0.55)};
    transform: none;
  }
`;

export const BirdsFlying = styled.img`
  position: absolute;
  top: 30vh;
  transform: translate(-4%, 0);
  width: 200px;
  opacity: 0.3;
  animation-name: ${Navigating};
  animation-duration: 120s;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
  animation-fill-mode: forwards;
  filter: contrast(0.5) opacity(0.6);

  @media all and (min-width: 768px) {
    top: 20%;
  }
  @media all and (min-width: 2000px) {
    top: 33%;
    width: 350px;
  }
`;
BirdsFlying.displayName = 'BirdsFlying';

export const KindFish = styled.img`
  display: inline-block;
  width: 80px;
  margin-right: -25px;
  vertical-align: top;
`;
KindFish.displayName = 'KindFish';

export const InfoArea = styled.div`
  z-index: 10;
  height: 230px;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;

  @media all and (min-width: 500px) {
    align-items: flex-start;
    height: 400px;
    width: 70%;
    padding: 10px 20px 10px 40px;
  }
`;
InfoArea.displayName = 'Info Area';

export const InfoMessage = styled.div`
  display: inline-block;
  margin-top: 20px;
  width: 90%;
  font-size: 0.9rem;
  line-height: 1.2;
  color: ${colorBlueLight};
  text-align: center;
  animation: ${Disappear} 10s;
  animation-fill-mode: forwards;

  @media all and (min-width: 500px) {
    animation: none;
    margin: 20px 0 0 20px;
    height: 200px;
    width: 80%;
    font-size: 1.2rem;
    line-height: 1.5;

    text-align: left;
  }

  @media all and (min-width: 2000px) {
    font-size: 1.4rem;
  }
`;
InfoMessage.displayName = 'Info Message';

export const JobTitle = styled.h2`
  width: 42vw;
  margin-top: -15px;
  font-family: ${fontSubtitle};
  font-size: 1.3rem;
  text-align: center;

  @media all and (min-width: 500px) {
    margin-top: -20px;
    margin-left: 15px;
    width: auto;
    font-size: 2rem;
    text-align: left;
  }
`;
JobTitle.displayName = 'Job Title';

export const LineOfArrows = styled.div`
  opacity: 0;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  animation-duration: 8s;
  animation-iteration-count: infinite;
  animation-fill-mode: forwards;
  width: 125%;

  &:nth-of-type(1) {
    animation-delay: 2s;
  }
  &:nth-of-type(2) {
    animation-delay: 3s;
  }
  &:nth-of-type(3) {
    animation-delay: 4s;
  }

  @media all and (min-width: 768px) {
    justify-content: center;
    width: auto;
    animation-name: ${DownWeGo(30)};
  }
  @media all and (min-width: 1400px) {
    padding-top: 5vh;
    animation-name: ${DownWeGo(50)};
  }
  @media all and (min-width: 2000px) {
    padding-top: 5vh;
    animation-name: ${DownWeGo(100)};
  }
`;
LineOfArrows.displayName = 'Line of Arrows';

export const MainAreaWrapper = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: flex-start;
  flex-wrap: wrap;
  width: 45%;

  @media all and (min-width: 500px) {
    width: 75%;
    flex-direction: row;
  }
`;
MainAreaWrapper.displayName = 'Main Area';

export const MeuBarquinho = styled.img`
  position: absolute;
  top: 40%;
  filter: saturate(0.5) brightness(1.8) contrast(30%) opacity(0.2);
  transform: translate(-50%, 0);
  width: 100px;
  animation-name: ${Navigating};
  animation-duration: 120s;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
  animation-fill-mode: forwards;

  @media all and (min-width: 768px) {
    top: 30%;
    filter: saturate(0.5) brightness(1.8) contrast(30%) opacity(0.2);
  }

  @media all and (min-width: 2000px) {
    top: 40%;
    width: 200px;
  }
`;
MeuBarquinho.displayName = 'Ship Image';

export const MoreText = styled.p`
  z-index: 50;
  position: absolute;
  top: 50%;
  transform: translate(0, -50%);
  font-family: ${fontTitleAlt};
  color: ${colorWhite};
`;
MoreText.displayName = 'MoreText';

export const MyInfoInnerWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: flex-start;
  flex-wrap: wrap;
  margin: 0 auto;
  width: 95%;

  @media all and (min-width: 500px) {
    width: 90%;
  }
`;
MyInfoInnerWrapper.displayName = 'Inner Wrapper';

export const MyInfoWrapper = styled.div`
  height: ${`${window.innerHeight}px`};
  margin-top: 50px;
  width: 100%;

  @media all and (min-width: 500px) {
    height: 90vh;
    margin-top: 100px;
  }
`;
MyInfoWrapper.displayName = 'My Info Area';

export const MyNameOnDesktopDevices = styled(MyName)`
  ${NotDisplayed};
  text-align: left;
  @media all and (min-width: 500px) {
    display: block;
  }

  @media all and (min-width: 500px) {
    display: block;
  }
`;
MyNameOnDesktopDevices.displayName = 'My Name';

export const MyNameOnMobileDevices = styled(MyName)`
  display: block;
  margin: 10px 0 5px;
  text-align: center;

  @media all and (min-width: 500px) {
    ${NotDisplayed};
  }
`;
MyNameOnMobileDevices.displayName = 'My Name';

export const Photo = styled.img`
  margin-top: -15%;
  width: 100%;
  filter: contrast(1.4) opacity(0.65);

  @media all and (min-width: 768px) {
    filter: contrast(1.9) opacity(0.55);
    width: 130%;
  }
`;
Photo.displayName = 'Photo';

export const PhotoWrapper = styled.div`
  height: 170px;
  width: 100%;
  padding-top: 25px;
  height: auto;
  display: flex;
  align-items: flex-start;
  justify-content: center;

  @media all and (min-width: 500px) {
    height: 400px;
    width: 30%;
  }
`;
PhotoWrapper.displayName = 'Photo Wrapper';

interface ScrollDownDisplayProps extends React.ComponentPropsWithoutRef<'i'> {
  sections?: boolean;
}

export const ScrollDownDisplay = styled.div<ScrollDownDisplayProps>`
  z-index: 50;
  position: absolute;
  bottom: ${({ sections }) => (sections ? '10px' : '10vh')};
  right: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 0 10px;
  padding: 10px;
  height: ${({ sections }) => (sections ? '40px' : 'auto')};
  width: 55%;
  background: ${({ sections }) =>
    sections ? rgba(colorBlueDark, 0.6) : rgba(colorBlueDark, 0.4)};

  border-radius: 10px;

  @media all and (min-width: 768px) {
    bottom: 10%;
    right: unset;
    height: auto;
    width: 100%;
    align-items: flex-start;
    background: none;
    margin: 80px auto 0;
  }
`;
ScrollDownDisplay.displayName = 'ScrollDown Display';

export const SectionMyInfo = styled.section`
  ${MainBackground}
`;
SectionMyInfo.displayName = 'MY INFO Section';

export const SocialArea = styled.div`
  height: 400px;
  width: 55%;

  @media all and (min-width: 321px) {
    width: 50%;
  }

  @media all and (min-width: 500px) {
    width: 25%;
  }

  @media all and (min-width: 2000px) {
    height: 60vh;
  }
`;
SocialArea.displayName = 'Social Area';
