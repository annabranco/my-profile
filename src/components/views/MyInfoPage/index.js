import React from 'react';
import { Social } from '../';
import { Mugshot } from '../../../images';

export const MyInfoPage = ({ texts, textsSocial, viewedAll }) => {
  return (
    <section className="section__infoPage">
      <h1 className="infoPage__name infoPage__name--devices">Anna Branco</h1>

      <div className="infoPage__container">
        <div className="infoPage__photoAndName">
          <div className="infoPage__photo--container">
            <div className="infoPage__photo">
              <img
                src={Mugshot}
                alt="Anna Branco"
                className="infoPage__mugshot"
              />
            </div>
          </div>
          <div className="infoPage__id--container">
            <div className="infoPage__id">
              <h1 className="infoPage__name infoPage__name--desktop">
                Anna Branco
              </h1>
              <h2 className="infoPage__job">{texts.job}</h2>
            </div>
            <div className="infoPage__aditional">
              {!viewedAll ? texts.aditional : texts.thankyou}
            </div>
          </div>
        </div>
        <div className="infoPage__social--container">
          <Social texts={texts} />
        </div>
      </div>

      <img
        className="infoPage__boat"
        src="http://www.animatedimages.org/data/media/271/animated-ship-image-0031.gif"
        alt="Navigating beautifully"
      />

      {!window.matchMedia('(min-width: 1024px)').matches ? (
        <div className="infoPage__advise">{texts.advise}</div>
      ) : null}

      <div className="infoPage__scrollDown">
        <div className="infoPage__scrollDown-inner">
          <i className="fas fa-angle-double-down scroll-icon anim1"></i>
          <i className="fas fa-angle-double-down scroll-icon anim1"></i>
          <i className="fas fa-angle-double-down scroll-icon anim1"></i>
        </div>
        <div className="infoPage__scrollDown-inner">
          <i className="fas fa-angle-double-down scroll-icon anim2"></i>
          <i className="fas fa-angle-double-down scroll-icon anim2"></i>
          <i className="fas fa-angle-double-down scroll-icon anim2"></i>
        </div>
        <div className="infoPage__scrollDown-inner">
          <i className="fas fa-angle-double-down scroll-icon anim3"></i>
          <i className="fas fa-angle-double-down scroll-icon anim3"></i>
          <i className="fas fa-angle-double-down scroll-icon anim3"></i>
        </div>
      </div>
    </section>
  );
};
