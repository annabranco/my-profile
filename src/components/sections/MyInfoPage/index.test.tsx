import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import '../../../tests/mocks/matchMedia';
import MyInfoPage from '.';
import { setupMockProvider } from '../../../tests/mocks';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

describe('< MyInfoPage >', () => {
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    wrapper = shallow(
      <MockProvider>
        <MyInfoPage />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('should display my name correctly on desktop and devices', () => {});

  xit('should display my photo', () => {});

  xit('should display a job title updated with data obtained from server', () => {});

  xit('should render a Social component', () => {});

  xit('should display a thanks message when all app is viewed', () => {});

  xit('should display a boat animation', () => {});

  xit('should display a desktop recommendation advise on devices', () => {});

  xit('should display animated arrows on desktop', () => {});

  xit('should translate all texts when the language is changed', () => {});
});
