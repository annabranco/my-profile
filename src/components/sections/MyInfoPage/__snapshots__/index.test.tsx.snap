// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< MyInfoPage > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MyInfoPage />
</Provider>
`;
